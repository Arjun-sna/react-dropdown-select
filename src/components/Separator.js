import React from 'react';
import styled from '@emotion/styled';

const Separator = ({ props, state, methods }) =>
  props.separatorRenderer ? (
    props.separatorRenderer(props, state, methods)
  ) : (
    <SeparatorComponent className="react-dropdown-select-separator" />
  );

const SeparatorComponent = styled.div`
  border-left: 1px solid #ccc;
  width: 1px;
  height: 25px;
  display: block;
`;

export default Separator;
