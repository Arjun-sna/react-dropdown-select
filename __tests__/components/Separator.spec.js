import React from 'react';
import renderer from 'react-test-renderer';

import Separator from '../../src/components/Separator';

const props = {
  parentProps: {
    separatorRenderer: null
  }
};

it('<Separator/> renders correctly', () => {
  const tree = renderer.create(<Separator {...props}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
