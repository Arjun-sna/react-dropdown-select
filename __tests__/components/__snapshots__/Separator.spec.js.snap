// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Separator/> renders correctly 1`] = `
<div
  className="react-dropdown-select-separator css-vxbae9-SeparatorComponent e19h5j1v0"
/>
`;
